x-pack/plugins/ml/public/alerting/register_ml_alerts.ts

```diff
@@ -0,0 +1,95 @@
+      defaultMessage: 'Alert when anomaly detection jobs results match the condition.',
+    }),
+    iconClass: 'bell',
+    documentationUrl(docLinks) {
+      return `${docLinks.ELASTIC_WEBSITE_URL}guide/en/machine-learning/${docLinks.DOC_LINK_VERSION}/ml-configuring-alerts.html`;
+    },
+    alertParamsExpression: lazy(() => import('./ml_anomaly_alert_trigger')),
+    validate: (alertParams: MlAnomalyDetectionAlertParams) => {
+      const validationResult = {
+        errors: {
+          jobSelection: new Array<string>(),
+          severity: new Array<string>(),
+          resultType: new Array<string>(),
+        },
+      };
+
+      if (
+        !alertParams.jobSelection?.jobIds?.length &&
+        !alertParams.jobSelection?.groupIds?.length
+      ) {
+        validationResult.errors.jobSelection.push(
+          i18n.translate('xpack.ml.alertTypes.anomalyDetection.jobSelection.errorMessage', {
+            defaultMessage: 'Job selection is required',
+          })
+        );
+      }
+
+      if (alertParams.severity === undefined) {
+        validationResult.errors.severity.push(
+          i18n.translate('xpack.ml.alertTypes.anomalyDetection.severity.errorMessage', {
+            defaultMessage: 'Anomaly severity is required',
+          })
+        );
+      }
+
+      if (alertParams.resultType === undefined) {
+        validationResult.errors.resultType.push(
+          i18n.translate('xpack.ml.alertTypes.anomalyDetection.resultType.errorMessage', {
+            defaultMessage: 'Result type is required',
+          })
+        );
+      }
+
+      return validationResult;
+    },
+    requiresAppContext: false,
+    defaultActionMessage: i18n.translate(
+      'xpack.ml.alertTypes.anomalyDetection.defaultActionMessage',
+      {
+        defaultMessage: `Elastic Stack Machine Learning Alert:
+- Job IDs: \\{\\{#context.jobIds\\}\\}\\{\\{context.jobIds\\}\\} - \\{\\{/context.jobIds\\}\\}
+- Time: \\{\\{context.timestampIso8601\\}\\}
+- Anomaly score: \\{\\{context.score\\}\\}
+
+\\{\\{#context.isInterim\\}\\}
+Alerts are raised based on real-time scores. Remember that scores may be adjusted over time as data continues to be analyzed.
+\\{\\{/context.isInterim\\}\\}
+
+\\{\\{! Section might be not relevant if selected jobs don't contain influencer configuration \\}\\}
+Top influencers:
+\\{\\{#context.topInfluencers\\}\\}
+  \\{\\{influencer_field_name\\}\\} = \\{\\{influencer_field_value\\}\\} [\\{\\{score\\}\\}]
+\\{\\{/context.topInfluencers\\}\\}
+
+Top records:
+\\{\\{#context.topRecords\\}\\}
+  \\{\\{function\\}\\}(\\{\\{field_name\\}\\}) \\{\\{by_field_value\\}\\} \\{\\{over_field_value\\}\\} \\{\\{partition_field_value\\}\\} [\\{\\{score\\}\\}]
+\\{\\{/context.topRecords\\}\\}
+
+\\{\\{! Replace kibanaBaseUrl if not configured in Kibana \\}\\}
+[Open in Anomaly Explorer](\\{\\{\\{context.kibanaBaseUrl\\}\\}\\}\\{\\{\\{context.anomalyExplorerUrl\\}\\}\\})
```

Comment:
For an influencer type, I am getting an error on opening the Anomaly Explorer from the link (testing the email action)

![image](https://user-images.githubusercontent.com/7405507/107542932-9dc8af80-6bc0-11eb-9627-0482c6f24dea.png)

The link in the email is:
```
https://localhost:5601/abc/app/ml/explorer/?_g=(ml%3A(jobIds%3A!(ecommerce_high_sum_total_sales))%2Ctime%3A(from%3A%272021-02-10T08%3A00%3A00.000Z%27%2Cmode%3Aabsolute%2Cto%3A%272021-02-11T04%3A00%3A00.000Z%27))&_a=(explorer%3A(mlExplorerFilter%3A(filterActive%3A!t%2CfilteredFields%3A!(category.keyword%2C%27Women!%27s%20Clothing%2CWomen!%27s%20Shoes%27)%2CinfluencersFilterQuery%3A(bool%3A(minimum_should_match%3A1%2Cshould%3A!((match_phrase%3A(category.keyword%3A%27Women!%27s%20Clothing%2CWomen!%27s%20Shoes%27))))))%2CmlExplorerSwimlane%3A(selectedLanes%3A!(%27Women!%27s%20Clothing%2CWomen!%27s%20Shoes%27)%2CselectedTimes%3A1612980000%2CselectedType%3AviewBy%2CviewByFieldName%3Acategory.keyword)))
```

Reply:
Fixed in 399633657aa265c6cf4d1fcd689abf3083e0602d

Reply:
Testing https://github.com/elastic/kibana/pull/89286/commits/657b1fed5e449609ac3edb5ea7f838c23b05d35b the Anomaly Explorer is not loading correctly when clicking on a link in an email for a bucket alert. Link is:

```
https://localhost:5601/abc/app/ml/explorer/?_g=(ml%3A(jobIds%3A!(ecommerce_high_sum_total_sales))%2Ctime%3A(from%3A%272021-02-11T03%3A00%3A00.000Z%27%2Cmode%3Aabsolute%2Cto%3A%272021-02-11T23%3A00%3A00.000Z%27))&_a=(explorer%3A(mlExplorerFilter%3A()%2CmlExplorerSwimlane%3A(selectedLanes%3A!(Overall)%2CselectedTimes%3A1613048400%2CselectedType%3Aoverall)))
```

![image](https://user-images.githubusercontent.com/7405507/107627322-03677b00-6c57-11eb-93a0-087c41173df5.png)